compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3512,6 +3512,71 @@ TR::Register *OMR::Power::TreeEvaluator::vnegDoubleHelper(TR::Node *node, TR::Co
    return TR::TreeEvaluator::inlineVectorUnaryOp(node, cg, TR::InstOpCode::xvnegdp);
    }
 
+TR::Register *OMR::Power::TreeEvaluator::vabsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
+   switch(node->getDataType().getVectorElementType())
+     {
+     case TR::Int8:
+       return TR::TreeEvaluator::vabsIntHelper(node, cg, TR::InstOpCode::vspltisb, TR::InstOpCode::vsrab, TR::InstOpCode::vaddubm);
+     case TR::Int16:
+       return TR::TreeEvaluator::vabsIntHelper(node, cg, TR::InstOpCode::vspltish, TR::InstOpCode::vsrah, TR::InstOpCode::vadduhm);
+     case TR::Int32:
+       return TR::TreeEvaluator::vabsIntHelper(node, cg, TR::InstOpCode::vspltisw, TR::InstOpCode::vsraw, TR::InstOpCode::vadduwm);
+     case TR::Int64:
+       return TR::TreeEvaluator::vabsIntHelper(node, cg, TR::InstOpCode::vspltisw, TR::InstOpCode::vsrad, TR::InstOpCode::vaddudm);
+     case TR::Float:
+       return TR::TreeEvaluator::inlineVectorUnaryOp(node, cg, TR::InstOpCode::xvabssp);
+     case TR::Double:
+       return TR::TreeEvaluator::inlineVectorUnaryOp(node, cg, TR::InstOpCode::xvabsdp);
+     default:
+       TR_ASSERT_FATAL(false, "unrecognized vector type %s\n", node->getDataType().toString()); return NULL;
+     }
+   }
+
+TR::Register *OMR::Power::TreeEvaluator::vabsIntHelper(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic splt, TR::InstOpCode::Mnemonic shift, TR::InstOpCode::Mnemonic add)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *srcReg = cg->evaluate(firstChild);
+
+   TR::Register *resReg = cg->allocateRegister(TR_VRF);
+   TR::Register *shiftReg = cg->allocateRegister(TR_VRF);
+   TR::Register *maskReg = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(resReg);
+
+   //set shift amount to size of operand - 1
+   generateTrg1ImmInstruction(cg, splt, node, shiftReg, -1);
```

Comment:
`splt` doesn't need to be a variable here. For all four data types, `shiftReg` is filled with all 1s. You can can just pick one of the splat instructions to use and get rid of the `splt` parameter.